beacon_node/network/src/sync/manager.rs

```diff
@@ -895,6 +954,36 @@ impl<T: BeaconChainTypes> SyncManager<T> {
             RequestId::RangeBlockAndBlobs { id } => {
                 self.range_block_and_blobs_response(id, peer_id, blob.into())
             }
+            other => {
+                crit!(self.log, "Single blob received on incorrect request"; "request_id" => ?other);
+            }
+        }
+    }
+
+    fn rpc_data_column_received(
+        &mut self,
+        request_id: RequestId,
+        peer_id: PeerId,
+        data_column: Option<Arc<DataColumnSidecar<T::EthSpec>>>,
+        seen_timestamp: Duration,
+    ) {
+        match request_id {
+            RequestId::SingleBlock { .. } | RequestId::SingleBlob { .. } => {
+                crit!(self.log, "bad request id for data_column"; "peer_id" => %peer_id  );
+            }
+            RequestId::DataColumnsByRoot(id) => {
+                self.on_single_data_column_response(
+                    id,
+                    peer_id,
+                    match data_column {
+                        Some(data_column) => RpcEvent::Response(data_column, seen_timestamp),
+                        None => RpcEvent::StreamTermination,
+                    },
+                );
+            }
+            RequestId::RangeBlockAndBlobs { id } => {
```

Comment:
We should handle `RangeBlockAndDataColumns` here instead right?

Reply:
I think we should rename `RangeBlockAndBlobs` to `RangeBlockComponents` or `RangeBlockAndData` to be represent the set of sidecars the block needs. There's no need to have a separate enum variant. See
- https://github.com/sigp/lighthouse/pull/5561